src/leveldbwrapper.h

```diff
@@ -117,9 +145,11 @@ class CLevelDBWrapper
     template <typename K, typename V>
     bool Write(const K& key, const V& value, bool fSync = false) throw(leveldb_error)
     {
-        CLevelDBBatch batch;
-        batch.Write(key, value);
-        return WriteBatch(batch, fSync);
+        CLevelDBBatch* batch = GetBatch();
+        batch->Write(key, value);
+        bool written = WriteBatch(batch, fSync);
+        delete batch;
```

Comment:
As far as I can see it's not guaranteed that `batch` is deleted, namely if `WriteBatch -> HandleError -> throw`.

How about using something like [boost::scoped_ptr](http://www.boost.org/doc/libs/1_55_0/libs/smart_ptr/scoped_ptr.htm) to avoid the manual cleanup altogether?